py4web/core.py

```diff
@@ -953,20 +955,42 @@ def error404(error):
 # Web Server and Reload Logic: Operations
 #########################################################################################
 
+def watch_folder_event_loop(apps_folder):
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    loop.run_until_complete(watch_folder(apps_folder))
+
+async def watch_folder(apps_folder):
+    click.echo('watching python file changes in: %s' % apps_folder)
+    async for changes in awatch(os.path.join(apps_folder)):
+        for app in set([p.relative_to(apps_folder).parts[0]
+                for p in [pathlib.Path(pair[1]) for pair in changes]
+                if p.suffix == '.py']):
+            Reloader.import_app(app)
 
 def start_server(args):
-    host, port = args['host'], int(args['port'])
+    host, port, apps_folder, watch = args['host'], int(args['port']), args['apps_folder'], args['watch']
+
     if platform.system().lower() == "windows":
+        if watch:
+            # default wsgi server block the main thread so we open a new thread for the file watcher
+            threading.Thread(target=watch_folder_event_loop, args=(apps_folder,)).start()
         # Tornado fail on windows
         bottle.run(host=host, port=int(port), reloader=False)
     elif args['number_workers'] < 1:
+        if watch:
+            # tornado delegate to asyncio so we add a future into the event loop
+            asyncio.ensure_future(watch_folder(apps_folder))
         bottle.run(server="tornado", host=host, port=port, reloader=False)
     else:
         if not gunicorn:
             logging.error("gunicorn not installed")
         elif not gevent:
             logging.error("gevent not installed")
         else:
+            if watch:
+                click.echo('--watch option has no effect in multi-process environment \n')
+
```

Comment:
I don't really want to put the watch logic in start_server function. But the watcher initialization is affected by which server is used and I don't like to re-implement the same condition checking elsewhere either. Any advice on this?

Reply:
maybe place it in the separate module and do some like:
```python
def start_server(args):
    ...
   if watch and args['number_workers'] < 1: 
      import Watcher from whatever
      watcher = Watcher(Reloader, apps_folder, platform.system().lower())
  
```